lib/modules/iconv.c

```diff
@@ -82,7 +82,8 @@ static int iconv_convpath(struct iconv *ic, const char *path, char **newpathp,
 			if (!tmp)
 				goto err;
 
-			p = tmp + (p - newpath);
+			if (tmp != newpath)
+				p = tmp;
```

Comment:
Are you sure this is right? "p - newpath" looks like a pointer offset - calculate this before the realloc?

ptrdiff_t ptr_off = p - newpath;
tmp = realloc(newpath, newpathlen + 1);
...

p = tmp + ptr_off;